pkg/storage/etcd/etcd_watcher.go

```diff
@@ -398,10 +404,14 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 	var oldObj runtime.Object
 	if res.PrevNode != nil && res.PrevNode.Value != "" {
 		// Ignore problems reading the old object.
-		if oldObj, err = w.decodeObject(res.PrevNode); err == nil {
+		if oldObj, err = w.uncachedDecodeObject(res.PrevNode); err == nil {
 			oldObjPasses = w.filter(oldObj)
 		}
+		if err := w.versioner.UpdateObject(oldObj, res.Node.ModifiedIndex); err != nil {
```

Comment:
I'm not convinced this is the right fix. Given your PR description, I actually think sendModify was correct and sendDelete is doing the wrong thing-- it should send the RV corresponding with the object state, not a newer RV?